src/inode2prog.cpp

```diff
@@ -116,12 +116,32 @@ static std::string read_file(const char *filepath) {
   return contents;
 }
 
-std::string getprogname(pid_t pid) {
+std::string getcmdline(pid_t pid) {
   const int maxfilenamelen = 14 + MAX_PID_LENGTH + 1;
   char filename[maxfilenamelen];
 
   std::snprintf(filename, maxfilenamelen, "/proc/%d/cmdline", pid);
-  return read_file(filename);
+
+  bool replace_null = false;
+  std::string cmdline = read_file(filename);
+
+  // join parameters, keep prgname separate, don't overwrite trailing null
+  for (int idx = 0; idx < (cmdline.length() - 1); idx++) {
```

Comment:
Hmm, this gives a `comparison between signed and unsigned integer expressions` warning for me, can we avoid that?